src/packet/signature/types.rs

```diff
@@ -80,20 +80,25 @@ impl Signature {
         self.config.typ()
     }
 
+    /// Does `key` match any issuer or issuer_fingerprint subpacket in `sig`?
+    fn match_identity(sig: &Signature, key: &impl PublicKeyTrait) -> bool {
+        sig.issuer().iter().any(|&key_id| key_id == &key.key_id())
```

Comment:
One remark about this: before this patch, rpgp would validate signatures if *no* issuer subpackets existed in both areas.

This implementation does not. If no matching issuer or issuer fingerprint is found, it refuses to verify signatures.

In the tests we have, this doesn't seem to cause problems. However, I'm not entirely sure if there are edge cases that this PR breaks.

A possible addition that seems potentially attractive to me, would be to also return `true` in this helper if the hashed area contains neither an `Issuer` nor an `IssuerFingerprint` subpacket (in that case, the implementation that made the signature apparently did not want to put it on the record who issued the signature, but if we have an idea which key to verify it with, there is no downside to use trying).

(`ensure!` ing on `match_identity` is only an optimization that avoids spending the CPU cycles to check the signature. But at the risk that we reject a signature as not valid, even though it cryptographically is)

Reply:
After some more thought and discussion with @wiktor-k, I propose we tweak the logic as shown in 3c24ff17142511ae438ad9ed645fd16bd487f8f6 (i'll squash that commit up, if there is agreement)

Reply:
This should mostly align the semantics of this PR with the previous state, but gracefully handle the existence of multiple `Issuer` fields and `IssuerFingerprint`s.